### ISSUE TITLE
Tell us what you are using TotalSegmentator for

### ISSUE THREAD

**Issue by wasserth** [wasserth/TotalSegmentator#1]
It would be really helpful for us to now who is using TotalSegmentator and what they are using it for. Therefore, we invite you to comment on this issue: What are you using TotalSegmentator for? What classes are you using? Are there any classes you are missing (we can try to add them in future releases? Who are you (only if you want to disclose this)?

**Comment by Fivethousand5k** [comment]
我是同济伍谦兄

**Comment by LifeIsStrange** [comment]
@wasserth
HI sorry to disappoint, I'm am just an enthusiast amateur in medecine, I have no serious use for this software currently, still it is a potentially life saving advance in medecine.
Noob question:
I want to use the online version and give it as input a CT scan of a leg.
So my question is: what are the supported formats for the CT scan single file? 
I know it must be a zip but once uncompressed, can it be a regular image (JPEG, PNG) or does the image needs to conform to a specific format? 
E.g can I input this? 
https://www.researchgate.net/figure/Patient-Abdominal-CT-scan-Normal-diameter-appendix-with-no-evidence-of-inflammation-of_fig2_323371468/amp

**Comment by wasserth** [comment]
Hi,
the input needs to be a DICOM image (compressed inside of a zip) or a NIFTI image. JPEG or PNG does not work. You can find a example CT file here: https://github.com/wasserth/TotalSegmentator/tree/master/tests/reference_files

**Comment by LifeIsStrange** [comment]
Thank you

**Comment by rbumm** [comment]
Hi Jakob  @wasserth , 

Thank you for letting us try this great program. We will be using this for lung, lobe, and rib cage segmentation and add these as additional components for our surgical planning. 
Our project is to further improve our [lung segmentation and analysis platform](https://github.com/rbumm/SlicerLungCTAnalyzer) within 3D Slicer.   
Are there any plans to include segments, more detailed airways, or lung vessels in totalsegmentator? 

Best
Rudolf

**Comment by wasserth** [comment]
Great to hear that our tool is working well for you. We are currently working on adding airways and vessel segmentation for the lung as well as adding pneumothorax segmentation. What are you meaning by "segments"?

**Comment by wasserth** [comment]
Thanks for the explanation. We have no plans so far to add those segments. It seems that it would be quite a bit of effort to manually draw those segments since the borders are not clearly visible in the image (but are only defined by the vessels).

**Comment by rbumm** [comment]
Understand. How many datasets would you need for a sufficient segment training process with TotalSegmentator?

**Comment by wasserth** [comment]
In the beginning I would roughly need 20 labeled datasets. Then I would train a model and predict the 1200 subjects from the training dataset, then those predictions would need to be corrected where necessary.

**Comment by rbumm** [comment]
I'll discuss it with my team, but we could probably include this plan in our current 3D Slicer lung segmentation project and probably provide those 20 datasets within 6-8 weeks. 

Which data would you like to work on here for the initial training? Could we choose the data or would you define them?

**Comment by wasserth** [comment]
Sound good. Ideally you would pick 20 subjects from the TotalSegmentator dataset: https://zenodo.org/record/6802614
I does not really matter which subjects as long as they are from the training set (which subjects belong to which set can be seen in the `meta.csv` file which comes with the dataset).

**Comment by rbumm** [comment]
I'll contact you about this on Thursday. Which communication do you prefer? Email or are you on Discord?

**Comment by wasserth** [comment]
Email would be best: jakob.wasserthal@usb.ch

**Comment by natebraman** [comment]
What a phenomenal tool, thank you for making it publicly available. I am very interested in some of the additional models you have added after publication of the original paper, especially the vessel segmentation models since there's very little public data out there for this problem! 

Will you be posting metrics for these additional models, as you have for the [original 104 classes](https://github.com/wasserth/TotalSegmentator/blob/master/resources/results_all_classes.json)? Was the same training set used? Will you eventually be releasing the additional annotations?

**Comment by wasserth** [comment]
In the future we will also do evaluation for those additional models and release the labels. However, for now this is not very polished yet. For this reason I did not add them in the beginning, but after realizing that those models might already be quite helpful for other people I added them.

**Comment by mcastrorennes** [comment]
Hi, I found the solutioin very interessoing. I am evaluating the code. We want to use it to planning cardiac interventions.

**Comment by salimkanoun** [comment]
Hi, thank you for this amazing work, I was wondering if you thought about segmenting fat tissue (and differentiating perivisceral fat tissue from sub cutaneous fat tissue). we can consider making some annotation if you have interests in this project. Salim

**Comment by wasserth** [comment]
I am currently working on adding sub cutaneous fat and visceral fat segmentations. But it will take a few more weeks until it is done.

**Comment by salimkanoun** [comment]
That's cool, thank you for all the job done, really interesting project in which we hope to contribute in the future.

**Comment by kristianlindholm** [comment]
I'm currently working on a pre-clinical project at the University Hospital in Copenhagen, where we are interested in total tumor segmentation. We published an article about using the nnUnet for this task, with great results in improving time needed for physicians to do segmentations. Link to article: https://ejnmmires.springeropen.com/articles/10.1186/s13550-022-00901-2

Now we would like to combine the information of tumor segmentation with their location in the body, and I'll try to implement your organ segmentation model, to gain the knowledge of which organs the tumors are located in. This will allow physicians to quickly create an overview of the total tumor volume, as well as the location and volume of individual lesions. 

Thanks for the great work!! It's highly appreciated.
- Krsitian

**Comment by owensca** [comment]
I am going to test its results for segmenting ~150 whole-body CT scans of pediatric patients. My current interests are segmenting the lungs and possibly the bronchi and trachea. Any plans to train and validate this tool for segmenting CT scans of pediatric patients?

**Comment by LifeIsStrange** [comment]
@owensca I would love to read a blog about such real world uses.

**Comment by rbumm** [comment]
Lung lobe segmentation has recently become a 2-minute process with Lung CT Segmenter calls of the [TotalSegmentator AI extension](https://github.com/lassoan/SlicerTotalSegmentator).

Consequently, 3D Slicer Lung CT Analyzer 2.62 now includes lung lobe analysis of pulmonary infiltrates, lung collapse, and emphysema. A special emphysema result table and display are available which will be helpful in lung volume reduction surgery (LVRS) and the surveillance of COPD.

In this COPD case, 35% of the right upper lobe is affected, so this anatomical structure could be a primary target for LVRS.

![image](https://user-images.githubusercontent.com/18140094/208263215-3beef09a-76db-4980-b185-c112d3e614c6.png)

The pathology is well detected with conventional thresholding.  

https://user-images.githubusercontent.com/18140094/208263189-a09c2c60-b747-4152-a975-6bd086969d12.mp4

**Comment by smanwell** [comment]
Very excited to try out this tool, but I haven't yet had the chance to install the code. 

Can you please comment on the output format of TotalSegmentator? NIFTII? Volumetric or 2D?

Thank you!

**Comment by wasserth** [comment]
The output files are 3D volumetric nifti files.

**Comment by pangyuteng** [comment]
This is truly an impressive project, I can't even begin to fathom the amount of time and effort your team must have put into this, AND its all open source! 🙏🎉🎉🎉 

On top of my mind, I already know a few PIs that have existing research/cro projects that needed this YESTERDAY. Given that contouring of organs are done manually in many projects (typically oncology, with different organ focuses, liver, kidney...), any contour from CAD would be a productivity booster/tremendous time saver for the image annotation team.  Obviously, for clinical work, this will drop the turn around time for report generation/surgical planning, thus bringing quantitative image analyis one huge step towards clincal practice.

Below, I am sharing the dice computed from TotalSegmentator outputs versus manual contours in [AMOS2022](https://amos22.grand-challenge.org) (n=359) and [Pediatric-CT-SEG](https://wiki.cancerimagingarchive.net/pages/viewpage.action?pageId=89096588) datasets (also n=~359, age ranging from 0 to 16)   (** disclaimer, there could still be bugs in my assessment scripts, further no other datasets have 104 labels, thus i had to merge the 104 labels to the dataset's label set ** )

AMOS2022:
![img](https://raw.githubusercontent.com/pangyuteng/totalsegmentator-and-friends/main/assess/amoss22/results/summary-default.png){ width: 512px; }

Pediatric-CT-SEG (with and without --fast flag):
![img](https://raw.githubusercontent.com/pangyuteng/totalsegmentator-and-friends/main/assess/ped-ct-seg/results/compare.png){ width: 512px; }

Dice in Pediatric-CT-SEG group by age in liver (cc @owensca, since you mentioned you are interested using this for pediatric patients):
![img](https://raw.githubusercontent.com/pangyuteng/totalsegmentator-and-friends/main/assess/ped-ct-seg/results/age-vs-dice-Liver.png){ width: 512px; }

**Comment by philippenoelmarie** [comment]
Happy New year, congratulations on your amazing work. I am working as a doctor in the craniofacial area; it is already very useful to me that totalsegmentator give automatically the brain volume. I am more interested in the face, which appears as a global area. Is it a chance that it could be segmented in the future? I am not a geek, and I hardly understand the technical names, but I am a good clinician, and I wonder how I could help with the development of this part of the process. The skeletal part is easily done, but the segmentation of other parts: eyeball, masticatory muscles, pharyngeal muscles, would be useful in the clinical research on CFM malformations etc...

**Comment by Rz1010** [comment]
I would like to use this software to segment the spine and vertebra out of dicom images, but sadly the software is not working correctly (either with docker or normal build).

I would appreciate any help!

**Comment by wasserth** [comment]
Please open a new issue with your specific error message so we can try to fix your problem.

**Comment by wasserth** [comment]
> Happy New year, congratulations on your amazing work. I am working as a doctor in the craniofacial area; it is already very useful to me that totalsegmentator give automatically the brain volume. I am more interested in the face, which appears as a global area. Is it a chance that it could be segmented in the future? I am not a geek, and I hardly understand the technical names, but I am a good clinician, and I wonder how I could help with the development of this part of the process. The skeletal part is easily done, but the segmentation of other parts: eyeball, masticatory muscles, pharyngeal muscles, would be useful in the clinical research on CFM malformations etc...

Thanks for pointing to those structures. We will see if we can add the eyeball, masticatory muscles and pharyngeal muscles in a future release.

**Comment by philippenoelmarie** [comment]
It would be wonderful, if I could help in anyway you are welcome 🙏 Envoyé de mon iPhoneLe 3 janv. 2023 à 11:28, Jakob Wasserthal ***@***.***> a écrit :﻿

Happy New year, congratulations on your amazing work. I am working as a doctor in the craniofacial area; it is already very useful to me that totalsegmentator give automatically the brain volume. I am more interested in the face, which appears as a global area. Is it a chance that it could be segmented in the future? I am not a geek, and I hardly understand the technical names, but I am a good clinician, and I wonder how I could help with the development of this part of the process. The skeletal part is easily done, but the segmentation of other parts: eyeball, masticatory muscles, pharyngeal muscles, would be useful in the clinical research on CFM malformations etc...

Thanks for pointing to those structures. We will see if we can add the eyeball, masticatory muscles and pharyngeal muscles in a future release.

**Comment by mohamadamoud** [comment]
Hello,

I was wondering, can we specify the organs that we want segmented using Total Segmentator, or does it always have to perform total body segmentation.

**Comment by tomaroberts** [comment]
@wasserth and team – this work is really amazing. I applaud you for open-sourcing all of this and doing such a great job of making it easy to install and use. Well done.

We've done some early testing on data from our hospital and the results are very promising. We are developing an open-source AI deployment platform for the NHS in the UK, called AIDE. [Link here to recent press release.](https://www.aicentre.co.uk/news-and-events/news/aide-goes-open-source) Would love to show it to you and discuss in the context of TotalSegmentator. I'll drop you an email. Thanks again!

**Comment by FilippoCinotti** [comment]
@wasserth and team – thanks for this truly amazing work!
Is there any chance shoulder rotator cuff will be included in future updates of the algorithm?
There are several papers in literature referring to this structure as potentially really relevant on a correct shoulder pathology diagnosis.

**Comment by wasserth** [comment]
We will put the shoulder rotator cuff on our list for future releases.

**Comment by hannesUlrich** [comment]
hey @wasserth and team, 
thanks for this amazing tool. Yesterday a radiologist showed me your tool and I am fascinated by the results. I would like to use it for indexing CT images in terms of CBIT. I wonder if there is a way to output only (!) the statistics without writing the actual segmentations. Sure, i can just delete them later, but in terms of disk access this would be a desirable feature. 

Have you thought about something like this?

**Comment by wasserth** [comment]
I just added an argument `--skip_saving`. You have to install from master branch to use it. Statistics calculation will be a lot faster if you use the `--fast` option.

**Comment by scrachel** [comment]
Hi @wasserth and team, 
thanks a lot for your fantastic work. I'm an application manager at Bayer, working with Chinese radiologists on their academic projects. TotalSegmentator is a well-trained tool for segmentation and has helped me a lot in dealing with the weights of different nnunet models. And I've sent you an e-mail inquiring about the licenses. Looking forward to your reply.
Besides, I'm interested in segmenting the liver into healthy and diseased regions based on dual-energy CT images. What do you think of that idea?

**Comment by rbumm** [comment]
Dear Jakob, Hope you are fine. What is the status of the missing extremity bones right now ? Could you update us ?Best RudolfSent from my iPhoneOn 19 Jun 2023, at 10:14, Jakob Wasserthal ***@***.***> wrote:﻿
Great that our work is helpful for you. So far we have focused on segmenting only anatomical structures. Segmenting pathologys and diseased regions would open another big field. This is further in the future on our roadmap.

**Comment by QMPautomation** [comment]
Hi @wasserth and team,
Thanks for the time and the expertise dedicated to this project. I would like to use TotalSegmentator in a radiation therapy project as a contouring quality control tool. However, I noticed some common critical structures are not available in the current structure set. Do you have plans for adding prostate, seminal vesicles, breasts, and rectum?

**Comment by wasserth** [comment]
We have these on our agenda, but further in the future. They are not part of the next release.

**Comment by rsavjanimdphd** [comment]
> That is great news. If you want us to test any airways or vessels please let me know. We have a well-working airway segmentation tool at hand within 3D Slicer. 
> 
> In addition, we plan to set up an ML active learning platform using IDC data soon. This will be based on DICOMWeb, 3D Slicer, MONIALabel, and lungmask.  If we could also work with TotalSegmentator in this context it would be awesome. 
> 
> The lung segments (10 on the right, 8 on the left side) would be the next and final substructures of lobes - they are not clearly divided by structures like fissures but usually pyramids of tissues originating from the branches of segment bronchi and pulmonary arteries (not veins). The segments are clinically important because in lung surgery we have more and more switched to segmentectomies in recent years for small lung cancers. For that one needs to know whether or not the tumor is located centrally in one or more of these segments. Segment borders are always approximations on CT and can later be intraoperatively defined by lack of perfusion (if you cut the vessels) or lack of segmental ventilation (if you cut the bronchi).  
> 
> ![File:Lung-lobe-bronchus-segment.jpg](https://camo.githubusercontent.com/0cb33c20a8f9c25066b2e97a751a770303b0dcde25462430628689c7161fa274/687474703a2f2f7261646970656469612e636f6d2f57696b694d656469612f696d616765732f322f32332f4c756e672d6c6f62652d62726f6e636875732d7365676d656e742e6a7067)
> 
>  (wikipedia)

We would be interested in the 18 lung segmentation model, as well. Did you guys try segment some sample 20 training patients and try this already? Thanks!

**Comment by wasserth** [comment]
We did not further pursue this. So unfortunately we do not have such a model yet.

**Comment by bjw34032** [comment]
Excellent software! Any progress on the adipose tissue segmentations?  I noticed a post from November 2022.

**Comment by fohofmann** [comment]
Hi @wasserth and all,

thank you again for all your efforts, and I highly appreciate your great work. For an ongoing project we are using the vertebral body as a relative localization measure (e.g. at height / z-axis of L2 etc.). In this scenario, only the vertebral body matters but not the arch (consisting of procc. spinosi / procc. transversi).

As a workaround, one can segment the spine (using task=total), and remove the vertebral arch (by setting everything outside of the mask generated by task=vertebrae_body to 0). However, this requires an additional segmentation. I was wondering whether you might consider to create a new task like 'vertebrae_body_detailed'. This might not even require new segmentations but the training data could be generated by fusing the masks as described.

Even if you do not plan to do this, thank you and congratulations to TotalSegementator once again!
Best, Felix

**Comment by fohofmann** [comment]
`vertebrae_body` returns the vertebral bodies of the complete spine, without differentiating between individual vertebrae. as described one can use the vertebrae segmentations from `total` and combine that with `vertebrae body`, however this requires two segmentation jobs.

**Comment by ivanstepanovftw** [comment]
I am using TotalSegmentator to see my old medical imaging pictures without contrast and investigate my body

**Comment by gleesoi** [comment]
I would like to segment lungs,heart, breast/chest wall ctv/tissue and internal mammary nodes ctv for radiotherapy project. Any idea if those ctvs can be done or in future?thanks

**Comment by Peaceandmaths** [comment]
Hello, 

I am a Master student at University of Applied Science of Zurich (ZHAW) and using Total Segmentator to segment the brain from head-neck-torso CT images and further train nnunet on these images cropped to the ROI (brain) to segment intracranial aneurysms. The tool is working well, thank you very much ! 

I have a question, when I visualize the images in ITK-Snap  before and after applying TS + cropping they look quite different wrt the intensity range, i.e I see less details/nuances of grey. Is it normal ? Thanks in advance for any insights ! 


![image](https://github.com/wasserth/TotalSegmentator/assets/117741432/ada83d5a-dfd8-495a-b4fc-f2e5380ecc47)

![image](https://github.com/wasserth/TotalSegmentator/assets/117741432/f56c6d1b-c103-439f-a69d-05fd549fe168)

**Comment by NMontanaBrown** [comment]
Hi team,

We used TotalSegmentator to create a novel dataset called SARAMIS (Simulation Assets for Robotic Assisted and Minimally Invasive Surgery): https://github.com/NMontanaBrown/saramis and https://openreview.net/forum?id=SEU9m9NReo&noteId=SEU9m9NReo. It was presented at NeurIPS 23 Datasets and Benchmarks Track.

Thanks for making such a great tool!

**Comment by wasserth** [comment]
> Hello,
> 
> I am a Master student at University of Applied Science of Zurich (ZHAW) and using Total Segmentator to segment the brain from head-neck-torso CT images and further train nnunet on these images cropped to the ROI (brain) to segment intracranial aneurysms. The tool is working well, thank you very much !
> 
> I have a question, when I visualize the images in ITK-Snap before and after applying TS + cropping they look quite different wrt the intensity range, i.e I see less details/nuances of grey. Is it normal ? Thanks in advance for any insights !
> 
> ![image](https://private-user-images.githubusercontent.com/117741432/323398855-ada83d5a-dfd8-495a-b4fc-f2e5380ecc47.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ1OTAyMzEsIm5iZiI6MTcxNDU4OTkzMSwicGF0aCI6Ii8xMTc3NDE0MzIvMzIzMzk4ODU1LWFkYTgzZDVhLWRmZDgtNDk1YS1iNGZjLWYyZTUzODBlY2M0Ny5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwMVQxODU4NTFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01MzQ4ZDk1MWRkZWE3ZGJhZjhmZWJhMzI1OWQ2MTBiYmM1MTVkNzlkNTZkZDM4MzE0MWZiMGU4YTczNzg0N2NlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.npAIlxLKWctd5IAkrCebfqbmeRAO5AxiQGNEzi0_QJY)
> 
> ![image](https://private-user-images.githubusercontent.com/117741432/323398900-f56c6d1b-c103-439f-a69d-05fd549fe168.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ1OTAyMzEsIm5iZiI6MTcxNDU4OTkzMSwicGF0aCI6Ii8xMTc3NDE0MzIvMzIzMzk4OTAwLWY1NmM2ZDFiLWMxMDMtNDM5Zi1hNjlkLTA1ZmQ1NDlmZTE2OC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwMVQxODU4NTFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iMDIyNjlhYWJhZWE5NmE5YWFhZTQ0ZjUwMmZlMzQ2MjIxMDEzNDEyNDcyZDEzYjQxYWZkYjliYTJiMWY5NzMyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.-5nw75LDEDIgFW6YckfmN15aB3TitwiVF7BeEbapvcg)

Probably the autowindowing results in differnet contrast. Try to set a fixed window which is the same for both images.

**Comment by wasserth** [comment]
> I would like to segment lungs,heart, breast/chest wall ctv/tissue and internal mammary nodes ctv for radiotherapy project. Any idea if those ctvs can be done or in future?thanks

We are working on adding those. But it will take some more time.

**Comment by skill-diver** [comment]
Could you please add model weights for vertebrae class to do single part like vertebrae_C1 L1 T1 segementation? Thank you.

**Comment by skill-diver** [comment]
I know vertebrae_body could provide a whole spine segementation, but it could not give the subparts of C1 L1 T1 etc.

**Comment by jianjun0407** [comment]
I use the following command, https://github.com/wasserth/TotalSegmentator#run-via-docker, using docker, to call the algorithm.
I have two questions:
First, I found that the size is 27.2G. Is the parameter file of the model included in docker ?
![image](https://github.com/wasserth/TotalSegmentator/assets/19251556/1a9a80b5-cd39-4f4e-84c0-61e35177f65b)


Second, I do not find the model parameters file and config.json file on my local computer. If they are included in docker, how can I set "send_usage_stats" to False?

**Comment by ltminnovator** [comment]
Hi all,

Thank you so much for your hard work and for making your work so easy to deploy!

I'm a graduate student in a group working on ultrasound to CT registration, and we use your program to help identify landmark structures in the CT volumes.

One request: It appears that the base model does not have full coverage of the splenic and portal veins. I believe the team member responsible for this part of the project is merging their results from the full-body model and the liver vessels model, but would it be possible to create a version that is simply the organs and vessels in the stomach/liver region?

**Comment by hep-raidium** [comment]
Hi, 

I use TotalSegmentator in the context of my PhD. I help radiologists to establish quantitative biomarkers on patients' scanners. TS is very useful for this work: thank you very much! 

As the radiologists I work with are focused on hepatology diseases, I mainly used liver, liver segments, spleen and portal vein segmentations. I also use body composition biomarkers for a general aspect of the patient's state. I noticed that on this website : https://compute.totalsegmentator.com/volume-report, visceral fat can be segmented, whereas this structure is not documented on the repo github. Is there any reason for this? 
Would it be possible to add it ? 


Thanks again for providing this tool. 
Best, 

Hélène

**Comment by maartenterpstra** [comment]
Hi @wasserth,

Thank you for publishing the excellent TotalSegmentator model! We are using TotalSegmentator in the [SynthRAD2025](https://synthrad2025.grand-challenge.org/) challenge.

The goal of the challenge is to generate accurate synthetic CT (sCTs) images from either MRI or cone-beam CT (CBCT). As part of this challenge, we have ~900 paired MRI-CT and ~1500 CBCT-CT image pairs in the head-and-neck and abdomen/thorax region. The quality of sCTs is estimated using image similarity, geometric fidelity, and dose accuracy. We use TotalSegmentator to estimate the geometric fidelity. In particular, ground-truth segmentations are derived from the CT images using TotalSegmentator. Then, when participants submit their algorithm for evaluation, TotalSegmentator segments the sCTs on the fly during evaluation. The geometric fidelity is then estimated by computing the DICE and 95th percentile Hausdorff distance metrics between these segmentations.

To facilitate this, we had to modify the TotalSegmentator platform: we run the model inside a Docker container on a batch of images. We have modified the code so that the model weights are only loaded once and kept in memory to reduce the runtime of the evaluation container. Moreover, multiple sCTs are processed in parallel to maximize resource utilization and minimize runtime. With these changes, we find that TotalSegmentator provides accurate and reproducible segmentations, while keeping the runtime low when evaluating hundreds of cases.

Thank you for publishing the TotalSegmentator model, which is an integral part of the SynthRAD2025 challenge evaluation!

**Comment by eltzanis** [comment]
We’re excited to share [mAIstro](https://github.com/eltzanis/mAIstro), our open-source, modular, multi-agent system for medical image analysis!

mAIstro leverages the powerful TotalSegmentator for accurate and automatic segmentation in CT and MR images. We’ve developed a specialized TotalSegmentator Agent, which is integrated into our natural language-driven framework. Users can simply ask the Master Agent to segment specific organs or tissues or all available structures and the TotalSegmentator Agent is invoked to produce the segmentation masks automatically, without any coding or manual steps.

You can also create your own agents and add them to the team of task-specific agents that form the core of the multi-agent system we’ve developed.

Huge thanks to @wasserth and the TotalSegmentator team for this incredible tool that makes automation like this possible! 🙌

We’re happy to share [mAIstro](https://github.com/eltzanis/mAIstro) with the community, it’s fully open-source, flexible, and designed to be easily extended or customized to fit your specific needs.

Check it out here: https://github.com/eltzanis/mAIstro

**Comment by neko7055** [comment]
Hi,

Thank you so much for your dedication to maintaining this project! With PyTorch 2.4+ integrating native Intel GPU support (torch.xpu), developers can now use syntax that closely aligns with CUDA (e.g., device = torch.device("xpu"), tensor.to("xpu")). If feasible, I would be very grateful if TotalSegmentator could consider adding XPU support.

Thank you

**Comment by clicker71** [comment]
We run TotalSegmentator (craniofacial_structures) behind a standard DICOMweb + IHE AIW-I flow, no proprietary glue:

1. Any DICOMweb client (Weasis, curl, OHIF) STOWs a head/jaw CT.
2. The server mints an AIW-I-style UPS workitem per study.
3. A sidecar claims it, fetches the frames over WADO-RS, runs the task, and STOWs an RTSTRUCT (MANDIBLE / TEETH_LOWER / TEETH_UPPER) back into the same study.

All plain PS3.18 DICOMweb (QIDO/WADO/STOW/UPS) - no C-STORE, no DICOMDIR, no viewer plugins. Windows/Linux, ~2 MB, run guide inside; weights are not shipped:

https://github.com/clicker71/clarus-pacs/releases/tag/totalseg-demo-v0.1

Research demo, not for diagnostic use.
